org.eclipse.jdt.junit/src/org/eclipse/jdt/junit/launcher/JUnitLaunchConfigurationTab.java

```diff
@@ -1153,20 +1153,21 @@ public boolean isSelectedValid(Object element) {
 		ViewerFilter filter= new TypedViewerFilter(acceptedClasses) {
 			@Override
 			public boolean select(Viewer viewer, Object parent, Object element) {
-			    if (element instanceof IPackageFragmentRoot && ((IPackageFragmentRoot)element).isArchive())
-			        return false;
-			    try {
-					if (element instanceof IPackageFragment && !((IPackageFragment) element).hasChildren()) {
-						return false;
-					}
-				} catch (JavaModelException e) {
+				if (element instanceof IPackageFragmentRoot && ((IPackageFragmentRoot) element).isArchive())
 					return false;
-				}
 				return super.select(viewer, parent, element);
 			}
 		};
 
-		StandardJavaElementContentProvider provider= new StandardJavaElementContentProvider();
+		StandardJavaElementContentProvider provider= new StandardJavaElementContentProvider() {
+			@Override
+			public boolean hasChildren(Object element) {
+				if(element instanceof IPackageFragment) {
```

Comment:
What does this do?  I tried without it and it works the same.

Reply:
It fix a bug in UI. The packages show a button to expand but the contentprovider will never provide children for a package in this context.

![image](https://user-images.githubusercontent.com/3462260/217756671-6ac2d9f3-f741-4680-9486-b67dd50002d5.png)

Reply:
Ah, in the Configuration dialog.  Ok.